Formatting docs (#6)
StatistikStadtZuerich/sszvis: docs/static/style-guide/formats.md

> > This library provides various functions for formatting the data in the preferred style.

Reviewer: ```suggestion
> This library provides functions for formatting data following the SSZVIS style guide.
```

## Formatting Functions

#### `sszvis.formatText(x)`

Any given data is returned as a string and will be formatted as Text.

```table
rows:
  - Function: formatText
    Example: sszvis.formatText(16000)
    Result: "'16000'"
  - Function:
    Example: sszvis.formatText("Jahr")
    Result: "'Jahr'"
```

> #### `sszvis.formatNumber(x)`

Reviewer: Please also add some examples that show how numbers will be shortened if they have insignificant decimals. See `test/format.test.js` for more examples.

A number greater than 9'999 will have a thousands separator space.

```table
rows:
  - Function: formatNumber
    Example: sszvis.formatNumber(16000)
    Result: "16 000"
  - Function:
    Example: sszvis.formatNumber(5000)
    Result: "5000"
```

#### `sszvis.formatPreciseNumber(p, d)`

`p` = The desired precision
`d` = The number to be formatted

A number will be formatted to a particular precision. It is a function that accepts numbers and returns formatted values. This function is "curried", meaning that it is a function with multiple arguments, but when you call it with less than the full number of arguments, it returns a function that takes less arguments and has the arguments you did provide "pre-filled" as parameters.

> Note that preciseNumber(2, 14.1234) is equivalent to preciseNumber(2)(14.1234)

Reviewer: ```suggestion
Note that `preciseNumber(2, 14.1234)` is equivalent to `preciseNumber(2)(14.1234)`.
```

```table
rows:
  - Function: formatPreciseNumber
    Example: sszvis.formatPreciseNumber(2, 14.1234)
    Result: 14.12
  - Function:
    Example: sszvis.formatPreciseNumber(3, 14.1234)
    Result: 14.123
  - Function:

>     Example: sszvis.formatPreciseNumber(3)(14.1234)

Reviewer: This is duplicated

Author: Yes, but on purpose to demonstrate both ways to write this function.

    Result: 14.123
  - Function:
    Example: sszvis.formatPreciseNumber(0, 14.1234)
    Result: 14
```

#### `sszvis.formatPercent(x)`

> A number will be transformed into a percent format. Formats percentages on the range 0 - 100.

Reviewer: ```suggestion
A number will be transformed into a percent format. Formats percentages on the range 0–100.
```

```table
rows:
  - Function: formatPercent
    Example: sszvis.formatPercent(55.3)
    Result: "55.3 %"
  - Function:
    Example: sszvis.formatPercent(7)
    Result: "7 %"
  - Function:
    Example: sszvis.formatPercent(0.245875)
    Result: "0.25 %"
```

#### `sszvis.formatFractionPercent(x)`

> A number will be transformed into a percent format. Format percentages on the range 0 - 1.

Reviewer: ```suggestion
A number will be transformed into a percent format. Format percentages on the range 0–1.
```

```table
rows:
  - Function: formatFractionPercent
    Example: sszvis.formatFractionPercent(55.3)
    Result: "5530 %"
  - Function:
    Example: sszvis.formatFractionPercent(7)
    Result: "700 %"
  - Function:
    Example: sszvis.formatFractionPercent(0.245875)
    Result: "24.59 %"
```

#### `sszvis.formatAge(x)`

A number will be formatted as an age. Means that any decimal number will be rounded to a full number.

```table
rows:
  - Function: formatAge
    Example: sszvis.formatAge(55.54359375)
    Result: "56"
  - Function:

>     Example: sszvis.formatAge(555435)

Reviewer: This is not a very realistic age 😅

    Result: "555435"
```

#### `sszvis.formatYear(x)`

A date object will be formatted as a year. Gives the date's year.

```table
rows:
  - Function: formatYear
    Example: sszvis.formatYear(new Date())
    Result: "2020"
```

#### `sszvis.formatMonth(x)`

A date object will be formatted as a month. It will return a capitalized three-letter abbreviation of the German month name.

```table
rows:
  - Function: formatMonth
    Example: sszvis.formatMonth(new Date())
    Result: "JAN"
```

#### `sszvis.formatAxisTimeFormat(x)`

A multi time formatter used by the axis class.

```table
rows:
  - Function: formatAxisTimeFormat
    Example: sszvis.formatAxisTimeFormat(new Date())

>     Result: ".345"

Reviewer: This is an important function, but this example doesn't show what it does very well. Maybe we should not show an example and just say that it is used by d3 under the hood. (See https://github.com/d3/d3-time-format)

Author: Good input, but how do we explain when it can be used?

> This library provides various functions for formatting the data in the preferred style.

## Formatting Functions

#### `sszvis.formatText(x)`

Any given data is returned as a string and will be formatted as Text.

```table
rows:
  - Function: formatText
    Example: sszvis.formatText(16000)

>     Result: "'16000'"

Author: Used this to show that the returned format is a string. Not sure if I formatted this correctly.

  - Function:
    Example: sszvis.formatText("Jahr")
    Result: "'Jahr'"
```

#### `sszvis.formatNumber(x)`

A number greater than 9'999 will have a thousands separator space.

```table
rows:
  - Function: formatNumber
    Example: sszvis.formatNumber(16000)
    Result: "16 000"
  - Function:
    Example: sszvis.formatNumber(5000)
    Result: "5000"
```

#### `sszvis.formatPreciseNumber(p, d)`

`p` = The desired precision

> `d` = The number to be formatted

Author: Added these for a more detailed explanation of the function

Reviewer: I like these!
